### PR TITLE
Fix hadoop umask and document permissions

### PR DESCRIPTION
**Pull request by alexr-bq**
### Summary

Set hadoop configuration to not have a umask and document how to resolve permissions issues.

### Description

By default, hadoop uses a 0022 umask value, which messes with our connector where both sides need write permissions. This umask is a client-side value, so we disable it for our operation.

Additionally, documented the main ways to resolve permissions issues with the connector.

### Related Issue

VER-77200

### Additional Reviewers

@jonathanl-bq 
@NerdLogic

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ However you set up HDFS, Vertica needs to have a copy of the hadoop configuratio
 ALTER DATABASE <database name> SET HadoopConfDir = '/hadoop/conf/location/';
 ```
 
+
+
```
Comment (jonathanl-bq) [inline README.md:55]: Remove

**Review by jonathanl-bq — APPROVED** [review]

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala [hunk 2]
```diff
@@ -162,6 +162,7 @@ class HadoopFileStoreLayer(fileStoreConfig : FileStoreConfig, schema: Option[Str
   hdfsConfig.set(SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_WRITE.key, "CORRECTED")
   hdfsConfig.set(SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_READ.key, "CORRECTED")
   hdfsConfig.setEnum(ParquetOutputFormat.JOB_SUMMARY_LEVEL, JobSummaryLevel.NONE)
+  hdfsConfig.set(CommonConfigurationKeys.FS_PERMISSIONS_UMASK_KEY, "000")
```
Comment (alexey-temnikov) [inline connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala:165]: Can we add Unit or Integration Test for that?

File: examples/basic-read/src/main/scala/example/Main.scala [hunk 3]
```diff
@@ -27,6 +27,7 @@ object Main extends App {
     "db" -> conf.getString("functional-tests.db"),
     "staging_fs_url" -> conf.getString("functional-tests.filepath"),
     "password" -> conf.getString("functional-tests.password"),
+    "file_permissions" -> "777",
```
Comment (alexey-temnikov) [inline examples/basic-read/src/main/scala/example/Main.scala:30]: Will it work with 700 permissions?

Comment (alexr-bq) [reply examples/basic-read/src/main/scala/example/Main.scala:30]: Yes.

File: connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala (hunk 2 above)
Comment (alexr-bq) [reply connector/src/main/scala/com/vertica/spark/datasource/fs/FileStoreLayerInterface.scala:165]: Yeah I'll add an integration test for this.

**Pull request merged**